cloudinit/net/netplan.py

```diff
@@ -149,8 +164,10 @@ def _listify(obj, token=" "):
                     "via": route.get("gateway"),
                     "to": to_net,
                 }
-                if "metric" in route:
-                    new_route.update({"metric": route.get("metric", 100)})
+                # Priority for metric: 1. route's metric, 2. subnet's metric, 3. default 100
+                route_metric = route.get("metric")
+                if (route_metric or sn_metric) is not None:
+                    new_route.update({"metric": route_metric or sn_metric or 100})
```

Comment:
The parenethesis in this conditional do not make sense in all cases. Given that metric value can be 0 you'd have a falsy value for route_metric of 0 which will still leave you silently skipping preserving the requested `metric: 0` on the route.

Same with the new_route.update call below, relying on truthiness will be a pitfall if either route_metric or sn_metric are 0.

Maybe it's better to be explicit:
```suggestion
                if route_metric is not None:
                    new_route["metric"] = route_metric
                elif sn_metric is not None:
                    new_route["metric"] = sn_metric
                else:
                    new_route["metric"] = 100
```

Also I have dropped the default of 100. The only case we try to set metric on the new_route is if either route_metric or sn_metric are not None. If they are not-None we should expect to use the value provided at either scope and we won't be falling back to 100.